### PR TITLE
Now builds and uses deps in new mono-repo

### LINKED ISSUE #1: Do not refer to relative paths for dependencies
Now that the repo is public (and will probably be used by several peers for debugging), we probably want to refer to some specific dependencies (either through versioning or git revision).

This can possibly wait a bit until the debug-tools changes on eth-trie-utils / protocol-decoder are merged to their respective `main`.

### PR DESCRIPTION
**Pull request by BGluth**
Resolves #1.

Updated the deps so it no longer contains local `[patch]` sections and builds properly. Also updated the deps to point to the new mono repo [zk_evm](https://github.com/0xPolygonZero/zk_evm).

### CODE REVIEW

File: Cargo.toml
Comment (Nashtare) [inline Cargo.toml:10]: Shame on me, you should have told me you were thinking of exposing my silliness 😅 

Joke aside, this is blocked by the PR you mentioned + a new release.

**Review by Nashtare — APPROVED** [review]

File: Cargo.toml
Comment (BGluth) [reply Cargo.toml:10]: Haha don't worry, I can rewrite history. :zipper_mouth_face:

**Comment by BGluth** [comment]
Going to merge this, but going to first wait for the release of the next version of `zk_evm` (should be within the next day).

**Comment by BGluth** [comment]
Should be good to merge now. The fix upstream in `zk_evm` is still on `develop`, so once that is merged, we should replace the deps with `semver` again.

**Review by muursh — APPROVED** [review]
Let's do a `zk_evm` release. We're now able to prove 701/701 so it feels like a nice logical time.

**Pull request merged**
